File: java-client/src/main/java/com/bolour/boardgame/client/message/MachinePlayResponse.java
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */

package com.bolour.boardgame.client.message;

import com.bolour.boardgame.client.domain.GameMiniState;
import com.bolour.boardgame.client.domain.Piece;
import com.bolour.boardgame.client.domain.Point;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class MachinePlayResponse {

    public final GameMiniState gameMiniState;
    public final List<Piece> playedPieces;
    public final List<Point> deadPoints;

    @JsonCreator
    public MachinePlayResponse(
      @JsonProperty("gameMiniState") GameMiniState gameMiniState,
      @JsonProperty("playedPieces") List<Piece> playedPieces,
      @JsonProperty("deadPoints") List<Point> deadPoints) {
        this.gameMiniState = gameMiniState;
        this.playedPieces = playedPieces;
        this.deadPoints = deadPoints;
    }
}

File: scala-server/boardgamecommon/src/main/scala/com/bolour/boardgame/scala/common/message/MachinePlayResponse.scala
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *    https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */
package com.bolour.boardgame.scala.common.message

import com.bolour.boardgame.scala.common.domain.{GameMiniState, PlayPiece}
import com.bolour.plane.scala.domain.Point

/**
  * API response to a request to initiate a machine play.
  *
  * @param gameMiniState Mini-state of the game after the play.
  * @param playedPieces List of pieces and locations used in the machine play.
  * @param deadPoints Empty points on the board that have been determined to be
  *                   no longer playable after the machine's play was laid out on the board.
  */
case class MachinePlayResponse(
  gameMiniState: GameMiniState,
  playedPieces: List[PlayPiece],
  deadPoints: List[Point]
)
